_drafts/submitting-your-first-pull-request.markdown

```diff
@@ -0,0 +1,205 @@
+
+Before you can rebase, you first need to fetch the changes from the upstream (in this case HTBox). Run `git fetch HTBox`. The fetch itself won’t change your branch. It simply ensures that your local git repo has a copy of the changes from HTBox/master. Next, execute `git rebase HTBox/master`. This will rewind all your changes and then replay them after the changes that happened to HTBox/master.
+
+image
+
+Luckily, we had no merge conflicts to deal with here so we can proceed with pushing our changes up to GitHub with the `git push –f` command.
+
+image
+
+Now when we look at this branch on GitHub, we can see that it is no longer behind the HTBox/master branch.
+
+image
+
+Over time, you will also want to keep your master branch up to date. The process here is the same but you usually don’t need to use the force flag to push. The force flag is only necessary when you have made changes in that branch. 
+
+image
+
+> <i class="fa fa-warning" /> **Caution**: When you rebase, then push –f, you are rewriting the history for your branch. This normally isn’t a problem if you are the only person working on your branch. It can however be a big problem if you are collaborating with another developer on your branch. If you are collaborating with others, the merge approach is a safer option than the rebase option.
+
+## Dealing with Merge Conflicts
+
+Dealing with conflicts is the worst part of any source control system, including git. When I run into this problem I use a combination of the command line and the git tooling built in to Visual Studio. I like to use Visual Studio for this because the visualization used for resolving conflicts is familiar to me.
+
+If a merge conflict occurs during a rebase, git will spew out some info for you.
+
+image
+
+Don’t panic. What happens here is the rebase stops at the commit where the merge conflict happened. It is now up to you to decide how you want to handle this merge conflict. Once you have completed the merge, you can then continue the rebase by running the `git rebase –continue` command. Alternatively, you can cancel everything by running the `git rebase –abort` command.
+
+As I said earlier, I like to jump over to Visual Studio to handle the merge conflicts. In Visual Studio, with the solution file for the project open, open the file that has a conflict.
+
+image
+
+Here, we can see the conflicted area. You could merge it manually here, but there is a much better way. In Visual Studio, open the Team Explorer and select _Changes_.
+
+image
+
+Visual Studio knows that you are in the middle of a rebase and that you have conflicts.
+
+image
+
+Click the _Conflicts_ warning and then click the Resolve button to resolve merge conflicts for the conflicted file.
```

Comment:
When I read the first draft of your post, the language here didn't match the screenshot.  I think you called it the Resolve button, but in the screenshot it was a Merge button or something like that.